scripts/mirror-builtin-actors.sh

```diff
@@ -0,0 +1,114 @@
+            PUBLISHED_DATE=$(echo "$release" | jq -r '.published_at')
+            PUBLISHED_DATE_SEC=$(date -d "$PUBLISHED_DATE" +%s)
+
+            if echo "$TAG_NAME" | grep -qE '^v[0-9]+\.[0-9]+\.[0-9]+.*$' && [[ "$PUBLISHED_DATE_SEC" -ge "$TWO_YEARS_AGO" ]]; then
+                mkdir -p "$BASE_FOLDER/$TAG_NAME"
+            fi
+        done
+
+        page_url="$next_page_url"
+    done
+}
+
+fetch_and_process_releases "$API_URL"
+
+declare -a failed_uploads
+
+: 'Loop through each version directory to process and upload assets'
+while IFS= read -r version_dir; do
+    TAG_NAME=${version_dir#"$BASE_FOLDER"/}
+    VERSION_DIR="$version_dir"
+    if [ -d "$VERSION_DIR" ]; then
+        echo "Entering directory: $VERSION_DIR"
+
+        tag_url="$API_URL/tags/$TAG_NAME"
+        release=$(curl -sS "$tag_url")
+
+        # Check if the assets array is not null
+        if [[ $(echo "$release" | jq '.assets') != "null" ]]; then
+            ASSETS=$(echo "$release" | jq -c '.assets[]')
+
+            pushd "$VERSION_DIR" > /dev/null
+            echo "Processing assets for $TAG_NAME..."
+
+            echo "$ASSETS" | while IFS= read -r asset; do
+                DOWNLOAD_URL=$(echo "$asset" | jq -r '.browser_download_url')
+                FILE_NAME=$(echo "$asset" | jq -r '.name')
+
+                echo "Checking asset: $FILE_NAME"
+                if [ ! -f "$FILE_NAME" ]; then
+                    echo "Downloading $FILE_NAME..."
+                    wget -q "$DOWNLOAD_URL" -O "$FILE_NAME" || echo "Failed to download $FILE_NAME"
+                fi
+
+                echo "Checking $FILE_NAME against S3 version..."
+                TEMP_S3_DIR=$(mktemp -d)
+                s3cmd get --no-progress "s3://$BUCKET_NAME/$TAG_NAME/$FILE_NAME" "$TEMP_S3_DIR/$FILE_NAME" || true
+
+                if cmp --silent "$FILE_NAME" "$TEMP_S3_DIR/$FILE_NAME"; then
+                    echo "$FILE_NAME is the same in S3, skipping..."
+                else
+                    echo "Local $FILE_NAME is different. Uploading to S3..."
+                    if s3cmd --acl-public put "$FILE_NAME" "s3://$BUCKET_NAME/$TAG_NAME/$FILE_NAME"; then
+                        echo "Uploaded $FILE_NAME to s3://$BUCKET_NAME/$TAG_NAME/$FILE_NAME"
+                    else
+                        echo "Failed to upload $FILE_NAME. Logging to $FAILED_LOG"
+                        echo "$TAG_NAME/$FILE_NAME" >> "$FAILED_LOG"
+                        failed_uploads+=("$TAG_NAME/$FILE_NAME")
+                    fi
+                fi
+                rm -rf "$TEMP_S3_DIR"
+                rm -f "$FILE_NAME"
+            done
+            popd > /dev/null
+        else
+            echo "No assets found for $TAG_NAME."
+        fi
+    fi
+done < <(find "$BASE_FOLDER" -mindepth 1 -type d)
+
+: 'Send summary alert only if there were failed uploads'
+if [ ${#failed_uploads[@]} -ne 0 ]; then
+    send_slack_alert_with_failed
```

Comment:
Is this a function? Where is it declared?

Reply:
missed it from playing around with the script, but updated it now